metrics-operator/controllers/common/providers/random/random.go

```diff
@@ -0,0 +1,58 @@
+package random
+
+import (
+	"context"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/go-logr/logr"
+	metricsapi "github.com/keptn/lifecycle-toolkit/metrics-operator/api/v1alpha3"
+)
+
+type KeptnRandomProvider struct {
+	Log logr.Logger
+}
+
+// Every metric provider must implement this EvaluateQuery method.
+func (r *KeptnRandomProvider) EvaluateQuery(ctx context.Context, metric metricsapi.KeptnMetric, provider metricsapi.KeptnMetricsProvider) (string, []byte, error) {
+
+	// create a context for cancelling the request if it takes too long.
+	ctx, cancel := context.WithTimeout(ctx, 20*time.Second)
+	defer cancel()
+
+	// since our endpoint only returns a value when hitting /random route
+	query := "random"
+
+	result, err := r.Query(ctx, query)
+	if err != nil {
+		return "", nil, err
+	}
+	return "", result, nil
+}
+
+func (r *KeptnRandomProvider) Query(ctx context.Context, metric string) ([]byte, error) {
+
+	// create a new request with context
+	request, err := http.NewRequestWithContext(ctx, http.MethodGet, "http://www.randomnumberapi.com/api/v1.0/random", nil)
+	if err != nil {
+		r.Log.Error(err, "Error in creating the request")
+		return nil, err
+	}
+
+	// make an http call using the default client.
+	response, err := http.DefaultClient.Do(request)
```

Comment:
for testing purposes, it is better to pass an HTTP client when creating the `KeptnRandomProvider`. This way, we can pass through a "fake" implementation to facilitate unit tests. See https://github.com/keptn/lifecycle-toolkit/blob/main/metrics-operator/controllers/common/providers/prometheus/prometheus_test.go#L71 for an example